contracts/core/Allo.sol

```diff
@@ -1,89 +1,270 @@
     }
 
-    // call to payoutStrategy.distribute()
-    function distribute(uint _poolId, bytes memory _data) external {
-        // Implement the function here
+    // TODO: DO WE NEED THIS? Not every strategy needs this
+    /// @notice Closes the pool
+    /// @param _poolId id of the pool
+    /// @dev Only callable by the pool owner
+    function closePool(uint256 _poolId) external {
+        if (!registry.isMemberOfIdentity(_identityId, msg.sender)) {
+            revert NO_ACCESS_TO_ROLE();
+        }
+        // pools[_poolId].allocationStrategy.close(); // ASK: Do we need this ?
+        pools[_poolId].distributionStrategy.close();
+        pools[_poolId].active = false;
+        emit PoolClosed(_poolId);
     }
 
-    // call to distributionStrategy.close()
-    function closePool(uint _poolId) external {
-        // Implement the function here
+    /// @notice Updates the treasury address
+    /// @param _treasury The new treasury address
+    /// @dev Only callable by the owner
+    function updateTreasury(address _treasury) external onlyOwner {
+        treasury = _treasury;
+        emit TreasuryUpdated(treasury);
     }
+
+    /// @notice Updates the fee
+    /// @param _fee The new fee
+    /// @dev Only callable by the owner
+    function updateFee(uint256 _fee) external onlyOwner {
+        fee = _fee;
+        emit FeeUpdated(fee);
+    }
+
+    /// @notice Create a determenstic clone of of contract
+    /// @param _contract The address of the contract to clone
+    function _createClone(address _contract) internal returns (address clone) {
+        require(_isContract(_contract), "not a contract");
+        bytes32 salt = keccak256(abi.encodePacked(msg.sender, _nonce++));
+        clone = ClonesUpgradeable.cloneDeterministic(_contract, salt);
+    }
+
+    /// @notice Checks if the address is a contract
+    /// @param _address The address to check
+    function _isContract(address _address) internal view returns (bool) {
+        uint256 codeSize;
+        assembly {
+            codeSize := extcodesize(_address)
+        }
+        return codeSize > 0;
+    }
+
+    /// @notice Transfers the amount to the address
+    /// @param _to The address to transfer to
+    /// @param _amount The amount to transfer
+    /// @param _token The address of the token to transfer
+    function _transferAmount(address payable _to, uint256 _amount, address _token) private {
+        if (_token == address(0)) {
+            // Native Token
+            (bool sent, bytes memory data) = _distributionStrategy.call{value: _amount}("");
+            if (!sent) {
+                revert TRANSFER_FAILED();
+            }
+        } else {
+            // ERC20 Token
+            IERC20Upgradeable(_token).safeTransfer(_to, _amount);
+        }
+    }
+
+    // ASK: Should registry be updatable
+    // ASK: Should we allow multiple registries
+    // ASK: Do we need closePool
+    // ASK: If yes -> do we need openPool to set active = true
```

Comment:
^ Open questions

Reply:
1. Should registry be updatable - Open
2. Should we allow multiple registries - Open
3. Do we need closePool - Removed `closePool()`
4. If yes -> do we need openPool to set active = true - Removed the `active` property from `Pool` struct

Reply:
- Added function to make registry updatable  
- Nope we stick to one registry (based on yesterdays calls)